### ISSUE TITLE
indent: SwitchCase default changed

### ISSUE THREAD

**Issue by seragh** [eslint-stylistic/eslint-stylistic#1123]
### Validations

- [x] Follow our [Code of Conduct](https://github.com/eslint-stylistic/eslint-stylistic/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://eslint.style/contribute/guide).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.

### Describe the bug

The default value for the option SwitchCase was 0. Then stylistic came along and I migrated. It continued to behave as if it was 0. Then with a recent eslint update it suddenly is 1.

Guessing the cause, in stylistic the value was changed to 1 quite a while back and an older eslint suppressed the new value for the time being and so the issue only surfaced just now.

This is technically a regression, and if you want to keep the new default, then please update the documentation accordingly.

### Reproduction

-

### System Info

```shell
-
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests
